tests/test_recursive.py

```diff
@@ -79,14 +78,28 @@ def __init__(self):
 def test_partials():
     assert dill.copy(SubMachine(), byref=True)
     assert dill.copy(SubMachine(), byref=True, recurse=True)
-   #if not OLDER: #FIXME: fails __main__.py
-   #    assert dill.copy(SubMachine(), recurse=True)
+    assert dill.copy(SubMachine(), recurse=True)
     assert dill.copy(SubMachine())
 
 
+class obj4(object):
+    def __init__(self):
+        super(obj4, self).__init__()
+        a = self
+        class obj5(object):
+            def __init__(self):
+                super(obj5, self).__init__()
+                self.a = a
+        self.b = obj5()
+
+
+def test_circular_reference():
+    assert dill.copy(obj4())
+
 
 if __name__ == '__main__':
     #print(('byref','_super','_recurse','_memo','_stop','OLDER'))
     test_super()
     test_partial()
     test_partials()
+    test_circular_reference()
```

Comment:
it's probably worthwhile to visit any of the current open issues were `super` is the key blocker, and add a test case when relevant.  May be able to also close those issues.